rule_packages/c/Declarations1.json

```diff
@@ -0,0 +1,140 @@
+              "description": "This query does not check for implicit function declarations as this is partially compiler checked.",
+              "items": []
+            }
+          }
+        ],
+        "title": "Declare identifiers before using them"
+      },
+      "DCL37-C": {
+        "properties": {
+          "obligation": "rule"
+        },
+        "queries": [
+          {
+            "description": "Declaring a reserved identifier can lead to undefined behaviour.",
+            "kind": "problem",
+            "name": "Do not declare or define a reserved identifier",
+            "precision": "very-high",
+            "severity": "warning",
+            "short_name": "DoNotDeclareOrDefineAReservedIdentifier",
+            "shared_implementation_short_name": "DeclaredAReservedIdentifier",
+            "tags": [
+              "correctness",
+              "maintainability",
+              "readability"
+            ],
+            "implementation_scope": {
+              "description": "This query does not consider identifiers described in the future library directions section of the standard. This query also checks for any reserved identifier as declared regardless of whether its header file is included or not.",
+              "items": []
+            }
+          }
+        ],
+        "title": "Do not declare or define a reserved identifier"
+      }
+  },
+  "MISRA-C-2012": {
+    "RULE-21-2": {
+      "properties": {
+        "obligation": "required"
+      },
+      "queries": [
+        {
+          "description": "Declaring a reserved identifier can lead to undefined behaviour.",
+          "kind": "problem",
+          "name": "A reserved identifier or reserved macro name shall not be declared",
+          "precision": "very-high",
+          "severity": "warning",
+          "short_name": "DoNotDeclareAReservedIdentifier",
+          "shared_implementation_short_name": "DeclaredAReservedIdentifier",
+          "tags": [
+            "correctness",
+            "maintainability",
+            "readability"
+          ]
+        }
+      ],
+      "title": "A reserved identifier or reserved macro name shall not be declared"
+    },
+    "RULE-5-1": {
+      "properties": {
+        "obligation": "required"
+      },
+      "queries": [
+        {
+          "description": "Using nondistinct external identifiers results in undefined behaviour.",
+          "kind": "problem",
+          "name": "External identifiers shall be distinct",
+          "precision": "very-high",
+          "severity": "warning",
+          "short_name": "ExternalIdentifiersNotDistinct",
+          "tags": [
+            "correctness",
+            "maintainability",
+            "readability"
+          ],
+          "implementation_scope": {
+            "description": "This query checks the first 31 characters of identifiers as significant, as per C99. This query does not consider universal or extended source characters.",
```

Comment:
I think we should mention here that we do not find contraventions where the names are identical.

Reply:
done